bindings/rust/integration/tests/internet_kms_pq_client.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(feature = "network-tests")]
+mod pq_kms {
+    use s2n_tls::{config::Config, security::Policy};
+    use s2n_tls_tokio::TlsConnector;
+    use tokio::net::TcpStream;
+    use tracing::level_filters::LevelFilter;
+
+    #[derive(Debug)]
+    struct TestCase {
+        s2n_security_policy: &'static str,
+        expected_cipher: &'static str,
+        expected_kem: Option<&'static str>,
+    }
+
+    // When KMS moves to standardized ML-KEM, this test will fail. The test should
+    // then be updated with the new security policy that supports ML-KEM.
+    const TEST_CASES: &[TestCase] = &[
+        // positive case: negotiates kyber
+        TestCase {
+            s2n_security_policy: "KMS-PQ-TLS-1-0-2020-07",
+            expected_cipher: "ECDHE-KYBER-RSA-AES256-GCM-SHA384",
+            expected_kem: Some("kyber512r3"),
+        },
+        // negative cases: these policies support a variety of early kyber drafts.
+        // We want to confirm that non-supported kyber drafts successfully fall
+        // back to a full handshake.
+        TestCase {
+            s2n_security_policy: "KMS-PQ-TLS-1-0-2019-06",
+            expected_cipher: "ECDHE-RSA-AES256-GCM-SHA384",
+            expected_kem: None,
+        },
+        TestCase {
+            s2n_security_policy: "PQ-SIKE-TEST-TLS-1-0-2019-11",
+            expected_cipher: "ECDHE-RSA-AES256-GCM-SHA384",
+            expected_kem: None,
+        },
+        TestCase {
+            s2n_security_policy: "KMS-PQ-TLS-1-0-2020-02",
+            expected_cipher: "ECDHE-RSA-AES256-GCM-SHA384",
+            expected_kem: None,
+        },
+        TestCase {
+            s2n_security_policy: "PQ-SIKE-TEST-TLS-1-0-2020-02",
+            expected_cipher: "ECDHE-RSA-AES256-GCM-SHA384",
+            expected_kem: None,
+        },
+    ];
+
+    /// Purpose: ensure that we remain compatible with existing pq AWS deployments.
+    ///
+    /// This test makes network calls over the public internet.
+    ///
+    /// KMS is has PQ support. Assert that we successfully negotiate PQ key exchange.
+    // Everything is included in the same block to prevent "unused" errors when
+    // "network-tests" aren't enabled.
+    #[tokio::test]
+    async fn pq_kms_test() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Do we need to gate this to only when PQ is available somehow? Or are we requiring that the integration tests / network tests only ever be run with AWS-LC?

Reply:
> Do we need to gate this to only when PQ is available somehow?

Currently (in the ci) PQ is always available, because aws-lc is pulled in by default.

I wasn't planning on adding feature gating until we actually had non-pq configuration of the bindings running in CI, but let me know if you would like to see that as part of this PR. I'd likely just add a `pq` feature to the `integ` crate that would be enabled by default.

Reply:
Should we document the requirement then? Because future us might not expect the integ tests to only work on awslc.

Reply:
I added a an explicit PQ feature, and also added a "helper test" to provide a useful error and obvious failure if the tests are ever run with a libcrypto that doesn't support PQ.